### ISSUE TITLE
WFS exception is not treated properly

### ISSUE THREAD

**Issue by jmacura** [hslayers/hslayers-ng#4057]
# Bug

**Describe the bug**
When Layman (geoserver?) returns 200 OK, but the body contains Exception, it is not treated as such. E.g. when drawing into a layer which cannot be actually used for drawing (Postgres direct connection layer).

**To Reproduce**
1. Go to 'Datasources' and add a layer which cannot be used for drawing
2. Draw a new feature
3. See that the layer says it's "synchronizing" and the feature disappears
4. See error in the console

**Expected behavior**
The error message shall be propagated to the user and any code waiting to success shall finish (like layer synchronizing)

**Screenshots**
![image](https://github.com/hslayers/hslayers-ng/assets/5598693/3f6f6542-3bdf-445f-9beb-93b89cb5d667)

**Additional context**
This is an example exception response from Layman:
```
<?xml version="1.0" encoding="UTF-8"?><ows:ExceptionReport xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:ows="http://www.opengis.net/ows" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.0.0" xsi:schemaLocation="http://www.opengis.net/ows https://watlas.lesprojekt.cz/geoserver/schemas/ows/1.0.0/owsExceptionReport.xsd">
<ows:Exception exceptionCode="NoApplicableCode">
<ows:ExceptionText>{http://jmacura}polygonovy_test_postgres is read-only</ows:ExceptionText>
</ows:Exception>
</ows:ExceptionReport>
```

**Comment by FilipLeitner** [comment]
In this case, I'd say that the status code is incorrect eg. such a response should be handled differently on Layman.

**Comment by jmacura** [comment]
> In this case, I'd say that the status code is incorrect eg. such a response should be handled differently on Layman.

Might be. @fzadrazil mentioned that the exception was thrown directly by Geoserver, perhaps Layman only resent it.

**Comment by fzadrazil** [comment]
Ok, so it is definitely GeoServer error, Layman acts only as a proxy in this case. Althought it is most likely not valid by the HTTP specs to return status 200, it is how GeoServer treats errors and HSLayers should adapt. WMS spec is not clear in this, it only describes how the ExceptionReport should look like and suggests the server "may return other status codes" in some cases. Layman can modify the status code in the future to make it valid, but I suppose it will be the same issue even for non-Layman WFS layers (at least those published via GS).

**Comment by jmacura** [comment]
Cannot find a non-editable layer now to replicate.
Will change code to disable layer synchronization icon.

**Comment by jmacura** [comment]
Layer synchronization state fixed in f1014bb
